src/plus/webviews/graph/protocol.ts

```diff
@@ -37,10 +37,16 @@ export type GraphRefMetadata = RefMetadata | null;
 export type GraphUpstreamMetadata = UpstreamMetadata | null;
 export type GraphRefsMetadata = Record</* id */ string, GraphRefMetadata>;
 export type GraphHostingServiceType = HostingServiceType;
+export type GraphRefMetadataType = RefMetadataType;
 export type GraphMissingRefsMetadataType = RefMetadataType;
 export type GraphMissingRefsMetadata = Record</*id*/ string, /*missingType*/ GraphMissingRefsMetadataType[]>;
 export type GraphPullRequestMetadata = PullRequestMetadata;
 
+export const GraphRefMetadataTypes = {
+	Upstream: 'upstream',
+	PullRequest: 'pullRequests',
+};
```

Comment:
A note on this piece: I originally had it pulling these strings directly from the graph library. However, this caused the GitLens extension to fail to load entirely when debugging. I'm guessing it's because I was attempting to pull constants out of a typedef file, but not sure exactly why it failed.

Reply:
Yeah, can't use `const`s as values from d.ts files. d.ts are just types/shapes and are not available at runtime.

That's why https://github.dev/gitkraken/vscode-gitlens/blob/52d8b0ad6b5a0d33781410dae49af11c5fba1dc8/src/%40types/vscode.git.enums.ts#L1 exists, I had to separate out certain `const`s from the vscode.git.d.ts file so that they would get compiled in

Reply:
Thanks! We'll definitely want an enums export in the Graph in general then - I'll add that in as a separate task.

Reply:
I would also suggest changing this to:
```ts
export enum GraphRefMetadataTypes {
	Upstream = 'upstream',
	PullRequest = 'pullRequests',
}

export const supportedRefMetadataTypes: GraphRefMetadataType[] = Object.values(GraphRefMetadataTypes);
```

To avoid having to call `Object.values` each time when checking the metadata, and that will also make sure that enum stays in sync with the types, since the `supportedRefMetadataTypes` will fail if an enum/type value don't change together

Reply:
Done in commit: [Use enum and values array](https://github.com/gitkraken/vscode-gitlens/pull/2450/commits/ca61b441157e60ab0e166f148d9dce994b424b53)